Revert "Exit full screen on tab close" (fixes #2618)

benmosher opened the pull request:
Sorry, better PR. Found others that were based against `dev-channel`, that seems to be the right target.

Fixes #2618 by reverting commit 8cded9cfd080646b77d79e100cd7c3b6509217d4.

bbondy commented:
I think there is a bug relating to html5 video full screen. Could you try your patch with that too? 
Pls compare to what Chrome does for it.

bbondy commented:
You're still tweaking this one right?

benmosher commented:
I haven't had a chance to take a second look yet, no. I will need to get more familiar with the code to make an actual change, this is just a revert of your original commit for #2404.

benmosher commented:
Looking at this now. I think I see what needs to be done to satisfy #2404 and #2618.

benmosher commented:
@bbondy I'm not in love with how I solved this, but it does seem to work. 

I couldn't figure out how to do this from inside `windowStore`, as it seemed to be necessary to run JS in the `frame.webview` before closing it. Only other thing might be to handle it in `componentWillUnmount`?

bbondy commented:
This is a good idea but it's only working about half the time for me (race condition).
Could you add something to the `js/actions/webviewActions.js` w/ a callback and then call it from the windowStore close frame? Moving the rest of the window action close frame code inside the callback that you pass.  I think that would solve it consistently .

bbondy commented:
Note also that if it is not full screen to call the callback right away.

benmosher commented:
Ah, makes sense, didn't realize there were actions with access to the current `webview`. I'll take a look.

benmosher commented:
Updated. Seems to work as well. Removed from `componentWillMount`.

Composed action in windowActions.js's `closeFrame`, I'm skittish about importing `*Actions` inside a store, even if they don't dispatch anything... 😅

bbondy commented:
Excellent, thanks for working through this!
